File: Devart/dotConnect/PostgreSQL/Samples/General/VB/RowDataSet/RowDataSetDemoControl.vb
Imports Devart.Data.PostgreSql
Imports System
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Globalization
Imports System.Windows.Forms

Namespace Samples
  Public Class RowDataSetDemoControl
    Inherits BaseDemoControl

    ' Fields
    Private WithEvents command As Devart.Data.PostgreSql.PgSqlCommand
    Private WithEvents pgSqlDataAdapter As Devart.Data.PostgreSql.PgSqlDataAdapter
    Private WithEvents pgSqlCommandBuilder As Devart.Data.PostgreSql.PgSqlCommandBuilder
    Private WithEvents dataSet As System.Data.DataSet
    Private WithEvents splitter As System.Windows.Forms.Splitter
    Private WithEvents tbSql As System.Windows.Forms.TextBox
    Private WithEvents pnTop As System.Windows.Forms.Panel
    Private WithEvents btUpdate As System.Windows.Forms.Button
    Private WithEvents btClear As System.Windows.Forms.Button
    Private WithEvents btFill As System.Windows.Forms.Button
    Private WithEvents dataGrid As System.Windows.Forms.DataGrid

    ' Methods
    Public Sub New()
      Me.components = Nothing
      Me.InitializeComponent()
    End Sub

    Public Sub New(ByVal connection As PgSqlConnection)
      Me.New()
      Me.command.Connection = connection
      Me.tbSql.Text = Me.command.CommandText
    End Sub

    'UserControl overrides dispose to clean up the component list.
    Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
      If disposing Then
        If Not (components Is Nothing) Then
          components.Dispose()
        End If
      End If
      MyBase.Dispose(disposing)
    End Sub

    'Required by the Windows Form Designer
    Private components As System.ComponentModel.IContainer

    'NOTE: The following procedure is required by the Windows Form Designer
    'It can be modified using the Windows Form Designer.  
    'Do not modify it using the code editor.
    <System.Diagnostics.DebuggerStepThrough()> _
    Private Sub InitializeComponent()
      Me.command = New Devart.Data.PostgreSql.PgSqlCommand()
      Me.pgSqlDataAdapter = New Devart.Data.PostgreSql.PgSqlDataAdapter()
      Me.pgSqlCommandBuilder = New Devart.Data.PostgreSql.PgSqlCommandBuilder()
      Me.dataSet = New System.Data.DataSet()
      Me.splitter = New System.Windows.Forms.Splitter()
      Me.tbSql = New System.Windows.Forms.TextBox()
      Me.pnTop = New System.Windows.Forms.Panel()
      Me.btUpdate = New System.Windows.Forms.Button()
      Me.btClear = New System.Windows.Forms.Button()
      Me.btFill = New System.Windows.Forms.Button()
      Me.dataGrid = New System.Windows.Forms.DataGrid()
      CType(Me.dataSet, System.ComponentModel.ISupportInitialize).BeginInit()
      Me.pnTop.SuspendLayout()
      CType(Me.dataGrid, System.ComponentModel.ISupportInitialize).BeginInit()
      Me.SuspendLayout()
      '
      'command
      '
      Me.command.CommandText = "SELECT * FROM Dept_Container"
      Me.command.Name = "command"
      Me.command.Owner = Me
      '
      'pgSqlDataAdapter
      '
      Me.pgSqlDataAdapter.SelectCommand = Me.command
      '
      'pgSqlCommandBuilder
      '
      Me.pgSqlCommandBuilder.DataAdapter = Me.pgSqlDataAdapter
      Me.pgSqlCommandBuilder.UpdatingFields = ""
      '
      'dataSet
      '
      Me.dataSet.DataSetName = "NewDataSet"
      Me.dataSet.Locale = New System.Globalization.CultureInfo("en-US")
      '
      'splitter
      '
      Me.splitter.Dock = System.Windows.Forms.DockStyle.Top
      Me.splitter.Location = New System.Drawing.Point(0, 88)
      Me.splitter.MinExtra = 50
      Me.splitter.Name = "splitter"
      Me.splitter.Size = New System.Drawing.Size(376, 3)
      Me.splitter.TabIndex = 10
      Me.splitter.TabStop = False
      '
      'tbSql
      '
      Me.tbSql.Dock = System.Windows.Forms.DockStyle.Top
      Me.tbSql.Location = New System.Drawing.Point(0, 24)
      Me.tbSql.Multiline = True
      Me.tbSql.Name = "tbSql"
      Me.tbSql.Size = New System.Drawing.Size(376, 64)
      Me.tbSql.TabIndex = 9
      '
      'pnTop
      '
      Me.pnTop.Controls.Add(Me.btUpdate)
      Me.pnTop.Controls.Add(Me.btClear)
      Me.pnTop.Controls.Add(Me.btFill)
      Me.pnTop.Dock = System.Windows.Forms.DockStyle.Top
      Me.pnTop.Location = New System.Drawing.Point(0, 0)
      Me.pnTop.Name = "pnTop"
      Me.pnTop.Size = New System.Drawing.Size(376, 24)
      Me.pnTop.TabIndex = 8
      '
      'btUpdate
      '
      Me.btUpdate.FlatStyle = System.Windows.Forms.FlatStyle.System
      Me.btUpdate.Location = New System.Drawing.Point(152, 0)
      Me.btUpdate.Name = "btUpdate"
      Me.btUpdate.Size = New System.Drawing.Size(75, 23)
      Me.btUpdate.TabIndex = 2
      Me.btUpdate.Text = "Update"
      '
      'btClear
      '
      Me.btClear.FlatStyle = System.Windows.Forms.FlatStyle.System
      Me.btClear.Location = New System.Drawing.Point(76, 0)
      Me.btClear.Name = "btClear"
      Me.btClear.Size = New System.Drawing.Size(75, 23)
      Me.btClear.TabIndex = 1
      Me.btClear.Text = "Clear"
      '
      'btFill
      '
      Me.btFill.FlatStyle = System.Windows.Forms.FlatStyle.System
      Me.btFill.Location = New System.Drawing.Point(0, 0)
      Me.btFill.Name = "btFill"
      Me.btFill.Size = New System.Drawing.Size(75, 23)
      Me.btFill.TabIndex = 0
      Me.btFill.Text = "Fill"
      '
      'dataGrid
      '
      Me.dataGrid.AllowNavigation = False
      Me.dataGrid.BorderStyle = System.Windows.Forms.BorderStyle.None
      Me.dataGrid.CaptionVisible = False
      Me.dataGrid.DataMember = ""
      Me.dataGrid.DataSource = Me.dataSet
      Me.dataGrid.Dock = System.Windows.Forms.DockStyle.Fill
      Me.dataGrid.GridLineColor = System.Drawing.SystemColors.ActiveBorder
      Me.dataGrid.HeaderForeColor = System.Drawing.SystemColors.ControlText
      Me.dataGrid.Location = New System.Drawing.Point(0, 91)
      Me.dataGrid.Name = "dataGrid"
      Me.dataGrid.Size = New System.Drawing.Size(376, 147)
      Me.dataGrid.TabIndex = 11
      '
      'RowDataSetDemoControl
      '
      Me.Controls.Add(Me.dataGrid)
      Me.Controls.Add(Me.splitter)
      Me.Controls.Add(Me.tbSql)
      Me.Controls.Add(Me.pnTop)
      Me.Name = "RowDataSetDemoControl"
      Me.Size = New System.Drawing.Size(376, 238)
      CType(Me.dataSet, System.ComponentModel.ISupportInitialize).EndInit()
      Me.pnTop.ResumeLayout(False)
      CType(Me.dataGrid, System.ComponentModel.ISupportInitialize).EndInit()
      Me.ResumeLayout(False)
      Me.PerformLayout()

    End Sub

    ' Methods
    Private Sub btFill_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btFill.Click
      Me.pgSqlDataAdapter.Fill(Me.dataSet, "Table")
      Me.dataGrid.DataMember = "Table"
    End Sub

    Private Sub btClear_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btClear.Click
      Me.dataGrid.DataMember = String.Empty
      Me.dataSet.Clear()
      Dim table As DataTable
      For Each table In Me.dataSet.Tables
        table.Constraints.Clear()
        table.Columns.Clear()
      Next
    End Sub

    Private Sub btUpdate_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btUpdate.Click
      If (dataSet.Tables.Count > 0) Then
        Me.pgSqlDataAdapter.Update(Me.dataSet, "Table")
      End If
    End Sub

    Private Sub tbSql_Leave(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles tbSql.Leave
      Me.command.CommandText = Me.tbSql.Text
      Me.pgSqlCommandBuilder.RefreshSchema()
    End Sub
  End Class
End Namespace

File: Devart/dotConnect/PostgreSQL/Samples/General/Backup/CS/RowDataSet/RowDataSetDemoControl.cs
using System;
using System.Data;

using System.Windows.Forms;


namespace Samples
{
	public class RowDataSetDemoControl : Samples.BaseDemoControl
	{
    private System.Windows.Forms.Splitter splitter;
    private System.Windows.Forms.DataGrid dataGrid;
    private System.Windows.Forms.TextBox tbSql;
    private System.Windows.Forms.Panel pnTop;
    private System.Windows.Forms.Button btUpdate;
    private System.Windows.Forms.Button btClear;
    private System.Windows.Forms.Button btFill;
    private Devart.Data.PostgreSql.PgSqlCommand command;
    private Devart.Data.PostgreSql.PgSqlDataAdapter pgSqlDataAdapter;
    private Devart.Data.PostgreSql.PgSqlCommandBuilder pgSqlCommandBuilder;
    private System.Data.DataSet dataSet;
		private System.ComponentModel.IContainer components = null;

		public RowDataSetDemoControl()
		{
			InitializeComponent();
  	}

    public RowDataSetDemoControl(Devart.Data.PostgreSql.PgSqlConnection connection) 
      : this() {

    this.command.Connection = connection;
    this.tbSql.Text = this.command.CommandText;
  }

		/// <summary>
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if (components != null) 
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		#region Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
      this.splitter = new System.Windows.Forms.Splitter();
      this.dataGrid = new System.Windows.Forms.DataGrid();
      this.dataSet = new System.Data.DataSet();
      this.tbSql = new System.Windows.Forms.TextBox();
      this.pnTop = new System.Windows.Forms.Panel();
      this.btUpdate = new System.Windows.Forms.Button();
      this.btClear = new System.Windows.Forms.Button();
      this.btFill = new System.Windows.Forms.Button();
      this.command = new Devart.Data.PostgreSql.PgSqlCommand();
      this.pgSqlDataAdapter = new Devart.Data.PostgreSql.PgSqlDataAdapter();
      this.pgSqlCommandBuilder = new Devart.Data.PostgreSql.PgSqlCommandBuilder();
      ((System.ComponentModel.ISupportInitialize)(this.dataGrid)).BeginInit();
      ((System.ComponentModel.ISupportInitialize)(this.dataSet)).BeginInit();
      this.pnTop.SuspendLayout();
      this.SuspendLayout();
      // 
      // splitter
      // 
      this.splitter.Dock = System.Windows.Forms.DockStyle.Top;
      this.splitter.Location = new System.Drawing.Point(0, 88);
      this.splitter.MinExtra = 50;
      this.splitter.Name = "splitter";
      this.splitter.Size = new System.Drawing.Size(376, 3);
      this.splitter.TabIndex = 7;
      this.splitter.TabStop = false;
      // 
      // dataGrid
      // 
      this.dataGrid.AllowNavigation = false;
      this.dataGrid.BorderStyle = System.Windows.Forms.BorderStyle.None;
      this.dataGrid.CaptionVisible = false;
      this.dataGrid.DataMember = "";
      this.dataGrid.DataSource = this.dataSet;
      this.dataGrid.Dock = System.Windows.Forms.DockStyle.Fill;
      this.dataGrid.GridLineColor = System.Drawing.SystemColors.ActiveBorder;
      this.dataGrid.HeaderForeColor = System.Drawing.SystemColors.ControlText;
      this.dataGrid.Location = new System.Drawing.Point(0, 88);
      this.dataGrid.Name = "dataGrid";
      this.dataGrid.Size = new System.Drawing.Size(376, 150);
      this.dataGrid.TabIndex = 8;
      // 
      // dataSet
      // 
      this.dataSet.DataSetName = "NewDataSet";
      this.dataSet.Locale = new System.Globalization.CultureInfo("en-US");
      // 
      // tbSql
      // 
      this.tbSql.Dock = System.Windows.Forms.DockStyle.Top;
      this.tbSql.Location = new System.Drawing.Point(0, 24);
      this.tbSql.Multiline = true;
      this.tbSql.Name = "tbSql";
      this.tbSql.Size = new System.Drawing.Size(376, 64);
      this.tbSql.TabIndex = 6;
      this.tbSql.Text = "";
      this.tbSql.Leave += new System.EventHandler(this.tbSql_Leave);
      // 
      // pnTop
      // 
      this.pnTop.Controls.AddRange(new System.Windows.Forms.Control[] {
                                                                        this.btUpdate,
                                                                        this.btClear,
                                                                        this.btFill});
      this.pnTop.Dock = System.Windows.Forms.DockStyle.Top;
      this.pnTop.Name = "pnTop";
      this.pnTop.Size = new System.Drawing.Size(376, 24);
      this.pnTop.TabIndex = 5;
      // 
      // btUpdate
      // 
      this.btUpdate.FlatStyle = System.Windows.Forms.FlatStyle.System;
      this.btUpdate.Location = new System.Drawing.Point(152, 0);
      this.btUpdate.Name = "btUpdate";
      this.btUpdate.TabIndex = 2;
      this.btUpdate.Text = "Update";
      this.btUpdate.Click += new System.EventHandler(this.btUpdate_Click);
      // 
      // btClear
      // 
      this.btClear.FlatStyle = System.Windows.Forms.FlatStyle.System;
      this.btClear.Location = new System.Drawing.Point(76, 0);
      this.btClear.Name = "btClear";
      this.btClear.TabIndex = 1;
      this.btClear.Text = "Clear";
      this.btClear.Click += new System.EventHandler(this.btClear_Click);
      // 
      // btFill
      // 
      this.btFill.FlatStyle = System.Windows.Forms.FlatStyle.System;
      this.btFill.Name = "btFill";
      this.btFill.TabIndex = 0;
      this.btFill.Text = "Fill";
      this.btFill.Click += new System.EventHandler(this.btFill_Click);
      // 
      // command
      // 
      this.command.CommandText = "SELECT * FROM Dept_Container";
      this.command.Name = "command";
      this.command.Owner = this;
      // 
      // pgSqlDataAdapter
      // 
      this.pgSqlDataAdapter.SelectCommand = this.command;
      // 
      // pgSqlCommandBuilder
      // 
      this.pgSqlCommandBuilder.DataAdapter = this.pgSqlDataAdapter;
      this.pgSqlCommandBuilder.UpdatingFields = "";
      // 
      // RowDataSetDemoControl
      // 
      this.Controls.AddRange(new System.Windows.Forms.Control[] {
                                                                  this.splitter,
                                                                  this.dataGrid,
                                                                  this.tbSql,
                                                                  this.pnTop});
      this.Name = "RowDataSetDemoControl";
      this.Size = new System.Drawing.Size(376, 238);
      ((System.ComponentModel.ISupportInitialize)(this.dataGrid)).EndInit();
      ((System.ComponentModel.ISupportInitialize)(this.dataSet)).EndInit();
      this.pnTop.ResumeLayout(false);
      this.ResumeLayout(false);

    }
		#endregion

    private void tbSql_Leave(object sender, System.EventArgs e) {
    
      command.CommandText = tbSql.Text;
      pgSqlCommandBuilder.RefreshSchema();
    }

    private void btFill_Click(object sender, System.EventArgs e) {
    
      pgSqlDataAdapter.Fill(dataSet, "Table");
      dataGrid.DataMember = "Table";
    }

    private void btClear_Click(object sender, System.EventArgs e) {
    
      dataGrid.DataMember = String.Empty;
      dataSet.Clear();
      foreach (DataTable table in dataSet.Tables) {
        table.Constraints.Clear();
        table.Columns.Clear();
      }
    }

    private void btUpdate_Click(object sender, System.EventArgs e) {
    
      if (dataSet.Tables.Count > 0)
        pgSqlDataAdapter.Update(dataSet, "Table");
    }
	}
}

